### PR TITLE
Inject dependencies into router and clean up a little

### PR DESCRIPTION
**Pull request by inverse**
- Inject all dependencies into router instead of relying on global objects
- Clean up code a little bit for readability

Do you guys have an opinion on using something like composer for autoloading/dependency management? 

Also what is your stance on adding tests?

### CODE REVIEW

File: route.php [hunk 1]
```diff
@@ -62,18 +92,33 @@ public function decode($q)
...
+        }
+        if (count($args) > 1) {
+            $this->action = $args[1];
+        }
+        if (count($args) > 2) {
+            $this->subaction = $args[2];
+        }
+
+        if (in_array($requestMethod, ['POST', 'DELETE', 'PUT'])) {
+            $this->method = $requestMethod;
```
Comment (inverse) [inline route.php]: I can't see where method is ever used in code? Should we just strip it out?

Comment (chaveiro) [inline route.php]: You have to take inconsideration all modules that might exist that are not part of core repo.

Comment (inverse) [inline route.php]: That makes sense, I won't change this behavior then :+1:

**Comment by glynhudson** [comment]
Thanks, is this PR now ready to be merged?

>Do you guys have an opinion on using something like composer for autoloading/dependency management?

No experience of using composer, sounds like a good idea. Does it work in a similar way to Ruby gems and bundler? What dependencies are you suggesting could be managed with composer?

**Comment by inverse** [comment]
Seems good from the testing I did. I haven't tried any of the other plugin though. Functionally it's the same though.

Totally the same - Just pull in 3rd party libraries and also handles autoloading - https://getcomposer.org/doc/01-basic-usage.md

**Comment by glynhudson** [comment]
Just tested this and it seems after applying your changes I could load load the [Wifi module](https://github.com/emoncms/wifi). Attempting  to navigate to `http://emonpi/emoncms/wifi`, just redirected me back to `http://emonpi/emoncms/feed/list`

**Comment by inverse** [comment]
Fixed :+1:

**Comment by glynhudson** [comment]
Working now 👍

**Pull request merged**